### DISCUSSION TITLE
call useAsyncData more than once in a page

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#27437] [Questions]
I have a product page and I fetch some data from shopify and I also use nuxt content to manage the markdown pages. I call `useAsyncData` twice in the same page. I am not sure if this is valid or if there is a better way to do this.

```html
<script setup>
const { handle } = useRoute().params

const { data, error } = await useAsyncData(`${handle}`, () => {
  return getProduct({
    variables: {
      handle
    }
  })
})

const { data: sizeGuide } = await useAsyncData(`${handle}-sizeGuide`, async () => {
  return queryContent('/size-guides').where({ sizeGuide: `${data?.value?.sizeGuide?.value}` }).findOne()
})
</script>
```

**Comment by xjccc** [comment]
Its sure for call `useAsyncData` anytimes in your pages.

And for the better way that. I think its a team identifies, everyone has different way for this. If team member abide by this, this is the better way.
